src/controller/CHIPDeviceController.cpp

```diff
@@ -1176,10 +1195,10 @@ void DeviceCommissioner::OnSessionEstablished()
 
     if (sendOperationalCertsImmediately)
```

Comment:
The flag here is "sendOperationalCertsImmediately", but it switches from NOCSR flow to DAC flow. Is there a way we can get around this?

Reply:
since DAC flow happens before the NOCSR flow according to the Spec, I'd suggest we rename the flag to 'sendCertificateChainImmediately'. Do you agree?

Reply:
It should be renamed `usingLegacyFlowWithImmediateStart`. That makes it clearer it needs a refactor. The else case with a state machine is the flow used over IP. Did you test over BLE only?

Reply:
Tested using IP in both manners:

    legacy way
    state machine way (--args='chip_use_clusters_for_ip_commissioning=true')